webflow-hacks/form-hacks/18/18.ts

```diff
@@ -0,0 +1,44 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const invalidDomains = ['gmail.com', 'yahoo.com', 'hotmail.com', 'competitor.com'];
+  const FORM_SELECTOR = '[fs-hacks-element="form"]';
+  const EMAIL_SELECTOR = '[fs-hacks-element="email"]';
+  const formElement = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  const emailInput = document.querySelector<HTMLInputElement>(EMAIL_SELECTOR);
+  if (!formElement || !emailInput) return;
+
+  formElement.addEventListener('submit', function (e) {
+    if (!validateEmail(emailInput.value, invalidDomains)) {
```

Comment:
I would move `invalidDomains` inside the `validateEmail` function, no need to pass it as an argument